src/lib/pubkey/pk_options.h

```diff
@@ -0,0 +1,197 @@
+/*
+* (C) 2024 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_PK_OPTIONS_H_
+#define BOTAN_PK_OPTIONS_H_
+
+#include <botan/pk_keys.h>
+#include <optional>
+#include <span>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace Botan {
+
+/**
+* Signature generation/verification options
+*
+* The normal usage of this is in a builder style, eg
+*
+* PK_Signature_Options()
+*   .with_hash("SHA-256")
+*   .with_der_encoded_signature()
+*   .with_context("Foo")
+*/
+class BOTAN_PUBLIC_API(3, 6) PK_Signature_Options {
+   public:
+      /// Create an empty PK_Signature_Options
+      ///
+      /// This can be further parameterized by calling with_xxx functions
+      PK_Signature_Options() = default;
+
+      PK_Signature_Options(PK_Signature_Options&& other) = default;
+      PK_Signature_Options& operator=(PK_Signature_Options&& other) = default;
+
+      PK_Signature_Options(const PK_Signature_Options&) = delete;
+      PK_Signature_Options& operator=(const PK_Signature_Options& other) = delete;
+      ~PK_Signature_Options();
```

Comment:
The moment we act on [the "pre-constructed hash object" notion](https://github.com/randombit/botan/pull/4318#issuecomment-2297682058), we will have a `std::unique_ptr<>` member in this class and the rule-of-zero will apply: you could remove all of them.